src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -175,4 +183,40 @@ private ProjectBag gather(Project project) {
         return new ProjectBag(projects, envelopes);
     }
 
+    public Page<Project> filterProjects(SearchFilter searchFilter, Pageable pageable) {
+        Query query = this.buildProjectsQuery(searchFilter);
+        log.debug("Project Search query: " + query.toString());
+
+        List<Project> projects = mongoTemplate.find(query.with(pageable), Project.class);
+        long count = mongoTemplate.count(query, Project.class);
+        return new PageImpl<>(projects, pageable, count);
+    }
+
+    public Query buildProjectsQuery(SearchFilter searchFilter) {
+        List<Criteria> criteria_list = new ArrayList<>();
+        criteria_list.add(Criteria.where("isUpdate").is(false));
+
+        Optional.ofNullable(searchFilter.getWranglingState())
+                .ifPresent(wranglingState -> {
+                    criteria_list.add(Criteria.where("wranglingState").is(wranglingState));
+                });
+
+        Optional.ofNullable(searchFilter.getWrangler())
+                .ifPresent(wrangler -> {
+                    criteria_list.add(Criteria.where("primaryWrangler").is(wrangler));
+                });
+
+        Optional.ofNullable(searchFilter.getSearch())
+                .ifPresent(search -> {
+                    // TextCriteria.forDefaultLanguage().matchingPhrase(searchPhrase);
+                    criteria_list.add(new Criteria().orOperator(
+                            Criteria.where("content.project_core.project_title").regex(search, "i"),
+                            Criteria.where("content.project_core.project_description").regex(search, "i"),
+                            Criteria.where("content.project_core.project_short_name").regex(search, "i")
```

Comment:
We need to keep an eye if ever these fields changed in the metadata schema in the future. I don't think they will, so this should be alright.